skyportal/handlers/api/followup_request.py

```diff
@@ -1694,3 +1705,124 @@ def delete(self, default_followup_request_id):
+
+            try:
+                followup_request_id = int(followup_request_id)
+            except ValueError:
+                return self.error("Assignment ID must be an integer.")
+
+            followup_requests = followup_requests.where(
+                FollowupRequest.id == followup_request_id
+            ).options(
+                joinedload(FollowupRequest.watchers),
+            )
+            followup_request = session.scalars(followup_requests).first()
+            if followup_request is None:
+                return self.error("Could not retrieve followup request.")
+
+            watchers = followup_request.watchers
+            if any([watcher.id == self.current_user.id for watcher in watchers]):
+                return self.error("User already watching this request")
+
+            watcher = FollowupRequestUser(
+                user_id=self.current_user.id, followuprequest_id=followup_request_id
+            )
+            session.add(watcher)
+            session.commit()
+
+            self.push_all(
+                action="skyportal/REFRESH_SOURCE",
+                payload={"obj_key": followup_request.obj.internal_key},
+            )
+
+            return self.success()
+
+    @auth_or_token
+    def delete(self, followup_request_id):
+        """
+        ---
+        description: Delete follow-up request from watch list
+        tags:
+            - followup_requests
+        parameters:
+            - in: path
+              name: followup_request_id
+              required: true
+              schema:
+                type: integer
+        responses:
+            200:
+               content:
+                application/json:
+                  schema: Success
+            400:
+              content:
+                application/json:
+                  schema: Error
+        """
+
+        with self.Session() as session:
+
+            # get owned assignments
+            followup_requests = FollowupRequest.select(self.current_user)
+
+            try:
+                followup_request_id = int(followup_request_id)
+            except ValueError:
+                return self.error("Assignment ID must be an integer.")
+
+            followup_requests = followup_requests.where(
+                FollowupRequest.id == followup_request_id
+            ).options(
+                joinedload(FollowupRequest.watchers),
+            )
+            followup_request = session.scalars(followup_requests).first()
+            if followup_request is None:
+                return self.error("Could not retrieve followup request.")
+
+            watcher = session.scalars(
+                FollowupRequestUser.select(self.current_user, mode="delete").where(
+                    FollowupRequestUser.user_id == self.current_user.id,
+                    FollowupRequestUser.followuprequest_id == followup_request_id,
+                )
+            ).first()
+            session.delete(watcher)
```

Comment:
Please verify that there was a result to this query, i.e that the user is actually a `watcher` of the followup request. If he isn't, it would make sense to return an error like `can't delete user {id} as he is not watching followup request {id} yet`.

Reply:
good point.